site/tests/app/libraries/FileUtilsTester.php

```diff
@@ -157,4 +157,204 @@ public function testGetAllDirs() {
         $this->assertEquals($folders, $dirs);
         FileUtils::recursiveRmdir($base_dir);
     }
+
+    private function buildFakeFile($filename, $part = 1, $err = 0) {
+        $_FILES["files{$part}"]['name'][] = $filename;
+        $_FILES["files{$part}"]['type'][] = FileUtils::getMimeType($this->path . $filename);
+        $_FILES["files{$part}"]['size'][] = filesize($this->path . $filename);
+
+        $tmpname = $this->path . Utils::generateRandomString() . $filename;
+        copy($this->path . $filename, $tmpname);
+
+
+        $_FILES["files{$part}"]['tmp_name'][] = $tmpname;
+        $_FILES["files{$part}"]['error'][] = $err;
+
+    }
+
+    private function fillToSize($fd, $target_size){
+        fseek($fd, $target_size-1,SEEK_CUR); 
+        fwrite($fd,'a'); 
+    }
```

Comment:
Let's combine this function with the one above as they are always used together.